patches/net/minecraft/client/gui/screens/MenuScreens.java.patch

```diff
@@ -23,3 +23,30 @@
      }
  
      @Nullable
+@@ -58,6 +_,11 @@
+         return (MenuScreens.ScreenConstructor<T, ?>)SCREENS.get(p_96200_);
+     }
+ 
++    /**
++     * @deprecated Use {@link net.neoforged.neoforge.client.event.RegisterMenuScreensEvent} to register menu screens
++     * TODO 1.20.5: remove AT related to this
++     */
++    @Deprecated
+     public static <M extends AbstractContainerMenu, U extends Screen & MenuAccess<M>> void register(
+         MenuType<? extends M> p_96207_, MenuScreens.ScreenConstructor<M, U> p_96208_
+     ) {
+@@ -65,6 +_,14 @@
+         if (screenconstructor != null) {
+             throw new IllegalStateException("Duplicate registration for " + BuiltInRegistries.MENU.getKey(p_96207_));
+         }
++    }
++
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public static void init() {
++        var map = new java.util.HashMap<MenuType<?>, ScreenConstructor<?, ?>>();
++        var event = new net.neoforged.neoforge.client.event.RegisterMenuScreensEvent(map);
```

Comment:
Why not pass the vanilla map directly? This would allow a single mod to override a vanilla screen, I don't think that is desirable.

Reply:
Yeah good point

Reply:
done!

Reply:
@Technici4n done!